pages/api/study/data/index.js

```diff
@@ -11,16 +11,40 @@ export default async function handler(req, res) {
   await dbConnect()
 
   try {
-    const { key } = req.body
+    const { key, ...participantData } = req.body
     const study = await Study.findOne({ key })
     if (study.data.length >= study.participantLimit) {
       return res.status(403).send()
     }
+    interpolateData(participantData)
     console.log(`Adding participant data for study with key ${key}...`)
-    const data = await ParticipantData.create({ ...req.body })
+    const data = await ParticipantData.create(participantData)
     await Study.findOneAndUpdate({ key }, { $push: { data: data._id } })
     res.status(201).json()
   } catch (error) {
     res.status(400).send()
   }
 }
+
+function interpolateData(participantData) {
```

Comment:
This might run slow. Probably won't be an issue since our server wont be used that much but some optimizations could def be made.
- instead of using array.find we could write our own search method leveraging the fact that the data is sorted by timestamp (binary search)
- instead of search thru array to see if it contains dataPoint, one pass aproach with 2 'pointers' and filling in gap between them could be more efficient

Reply:
Yeah it's definitely not optimal. Completing PR for now but keeping #18 open to work on optimization.